tools/src/main/java/org/apache/kafka/trogdor/coordinator/TaskManager.java

```diff
@@ -288,19 +289,44 @@ private JsonNode getCombinedStatus() {
     }
 
     /**
-     * Create a task.
+     * Creates and schedules the given tasks.
      *
-     * @param id                    The ID of the task to create.
-     * @param spec                  The specification of the task to create.
+     * @throws RequestConflictException - if a task with that ID and a different spec already exists
      */
-    public void createTask(final String id, TaskSpec spec)
-            throws Throwable {
-        try {
-            executor.submit(new CreateTask(id, spec)).get();
-        } catch (ExecutionException e) {
-            log.info("createTask(id={}, spec={}) error", id, spec, e);
-            throw e.getCause();
+    public void createAndScheduleTasks(Map<String, TaskSpec> givenTasks) {
+        Map<String, TaskSpec> tasks = new HashMap<>(givenTasks);
+        Iterator<Map.Entry<String, TaskSpec>> it = tasks.entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry<String, TaskSpec> idAndSpec = it.next();
+            String id = idAndSpec.getKey();
+            if (id.isEmpty())
+                throw new InvalidRequestException("Invalid empty ID in createTask request.");
+
+            TaskSpec spec = idAndSpec.getValue();
+            ManagedTask duplicateTask = this.tasks.get(id);
```

Comment:
`TaskManager#tasks` has to be accessed through the state change thread.  It can't be accessed here by incoming requests since there is no lock or synchronization.  Probably the easiest thing to do is to have a CreateTasks runnable which does what you want.